sdk/servicebus/azure-servicebus/azure/servicebus/_pyamqp/aio/_client_async.py

```diff
@@ -829,6 +830,48 @@ async def receive_message_batch_async(self, **kwargs):
             **kwargs
         )
 
+    async def receive_messages_iter_async(self, on_message_received=None):
+        """Receive messages by generator. Messages returned in the generator have already been
+        accepted - if you wish to add logic to accept or reject messages based on custom
+        criteria, pass in a callback.
+
+        :param on_message_received: A callback to process messages as they arrive from the
+         service. It takes a single argument, a ~uamqp.message.Message object.
+        :type on_message_received: callable[~uamqp.message.Message]
+        """
+        self._message_received_callback = on_message_received
+        return self._message_generator_async()
+
+    async def _message_generator_async(self):
+        """Iterate over processed messages in the receive queue.
+
+        :rtype: generator[~uamqp.message.Message]
+        """
+  
+        auto_complete = self.auto_complete
+        self.auto_complete = False
+        receiving = True
+        message = None
+        try:
+            while receiving:
+                receiving = await self.do_work_async()
+                message = self._received_messages.get(timeout=self._timeout)
+                if not self._received_messages.empty():
+                    self._received_messages.task_done()
+                yield message
+                await self._complete_message_async(message, auto_complete)
+        except:
+
+            if self._shutdown:
+                self.close()
+
+    async def _complete_message_async(self, message, auto):
+        if not message or not auto:
+            return
+        # this is off here, message delivery id?
+        await self.settle_messages_async(message[0][1], "accepted")
```

Comment:
this first parameter is the delivery ID, this needs to change. This function is also never used because we default pass in False for auto_complete into the receiever

Reply:
Should we remove altogether?